### PR TITLE
feat: PC-10671 Add Terraform for Honeycomb

### PR DESCRIPTION
**Pull request by BSski**
Besides the casual changes related to adding a new data source, a few other things landed in this PR:
- removal of "source_of" field for all data sources,
- nobl9-go update to version v0.73.0,
- small tweak for Azure Monitor: new "data_type" field.

I tested it on `dev-demo-2` using the following resources:
```terraform
resource "nobl9_project" "project" {
  display_name = "bsski-project-final Display"
  name         = "bsski-project-final"
  description  = "Lorem ipsum project"
}

resource "nobl9_service" "service" {
  display_name = "bsski-service-final Display"
  name         = "bsski-service-final"
  project      =  nobl9_project.project.name
  description  = "Lorem ipsum service"

  label {
    key    = "env"
    values = ["dev", "prod"]
  }

  label {
    key    = "team"
    values = ["red"]
  }
}

resource "nobl9_agent" "agent" {
  name      = "bsski-agent-final"
  project   =  nobl9_project.project.name
  agent_type = "honeycomb"
  release_channel = "beta"
  query_delay {
    unit = "Minute"
    value = 6
  }
}

resource "nobl9_direct_honeycomb" "direct" {
  name = "bsski-direct-final"
  project = nobl9_project.project.name
  description = "Lorem ipsum direct"
  api_key = "secret"
  log_collection_enabled = true
  release_channel = "beta"
  historical_data_retrieval {
    default_duration  {
      unit = "Day"
      value = 7
    }
    max_duration {
      unit = "Day"
      value = 7
    }
  }
  query_delay {
    unit = "Minute"
    value = 6
  }
}


resource "nobl9_slo" "slo" {
  display_name = "bsski-slo-final Display"
  name         = "bsski-slo-final"
  project      = nobl9_project.project.name
  service      = nobl9_service.service.name

  label {
    key = "team"
    values = ["green","sapphire"]
  }

  label {
    key = "env"
    values = ["dev", "staging", "prod"]
  }

  budgeting_method = "Occurrences"

  objective {
    display_name = "obj1"
    name = "tf-objective-1"
    target = 0.7
    value = 1
    op = "lt"
    raw_metric {
      query {
        honeycomb {
          calculation = "SUM"
          attribute = "test-column"
        }
      }
    }
  }

  time_window {
    count = 10
    is_rolling = true
    unit = "Minute"
  }

  indicator {
    name = nobl9_direct_honeycomb.direct.name
    project = nobl9_project.project.name
    kind = "Direct"
    }
}
```

### CODE REVIEW

File: nobl9/resource_direct.go [hunk 1]
```diff
@@ -22,29 +23,18 @@ type directResource struct {
...
-				Type:        schema.TypeList,
-				Required:    true,
-				MinItems:    1,
-				MaxItems:    2,
-				Description: "Source of Metrics and/or Services.",
-				Elem: &schema.Schema{
-					Type:        schema.TypeString,
-					Description: "Source of Metrics or Services.",
-				},
-			},
```
Comment (mkaras-nobl9) [inline nobl9/resource_direct.go:47]: issue: Removing the `source_of` field is a breaking change. Please use deprecation first 
image

We should deprecate this field before removal. - https://developer.hashicorp.com/terraform/plugin/framework/deprecations#provider-data-source-or-resource-removal

**Review by mkaras-nobl9 — CHANGES_REQUESTED** [review]
Honeycomb changes and API look good, but we should address issues of docs and `source_of` filed backward compatibility.

File: docs/resources/direct_honeycomb.md
[the remark this answers is not in the document]
Comment (BSski) [reply docs/resources/direct_honeycomb.md:1]: Thank you, fixed!

File: nobl9/resource_direct.go (hunk 1 above)
Comment (BSski) [reply nobl9/resource_direct.go]: Forgot about that, thanks! Fixed.

File: nobl9/resource_agent_test.go [hunk 2]
```diff
@@ -311,12 +299,27 @@ resource "nobl9_agent" "%s" {
...
+`, name, name, testProject)
+}
+
 func testGraphiteAgent(name string) string {
 	return fmt.Sprintf(`
 resource "nobl9_agent" "%s" {
   name      = "%s"
   project   = "%s"
-  source_of = ["Metrics", "Services"]
+
```
Comment (mkaras-nobl9) [inline nobl9/resource_agent_test.go:322]: nitpick: this line can be removed, right?

File: nobl9/resource_direct.go [hunk 3]
```diff
@@ -35,14 +36,15 @@ func resourceDirectFactory(directSpec directSpecResource) *schema.Resource {
...
 			"description":  schemaDescription(),
 			"source_of": {
+				Deprecated:  "source_of is Deprecated and not used anywhere. You can safely remove it from your configuration file.",
 				Type:        schema.TypeList,
-				Required:    true,
+				Optional:    true,
 				MinItems:    1,
 				MaxItems:    2,
-				Description: "Source of Metrics and/or Services.",
+				Description: "Deprecated value that was used to indicate whether the field was a source of Metrics and/or Services. It's not used anywhere but kept for backward compatibility.",
```
Comment (mkaras-nobl9) [inline nobl9/resource_direct.go:44]: nitpick:  It's not used anywhere **but, kept** for backward compatibility.

**Review by mkaras-nobl9 — DISMISSED** [review]
:1st_place_medal:

**Review by mkaras-nobl9 — APPROVED** [review]
:+1:

File: nobl9/resource_direct.go (hunk 3 above)
Comment (BSski) [reply nobl9/resource_direct.go]: As discussed on Slack - irrelevant after the new changes. Thanks!

**Review by mkaras-nobl9 — DISMISSED** [review]
:rocket: x :three:

**Pull request merged**